src/server/mcp.ts

```diff
@@ -617,13 +713,67 @@ export class McpServer {
     }
 
     const cb = rest[0] as PromptCallback<PromptArgsRawShape | undefined>;
-    this._registeredPrompts[name] = {
+    const registeredPrompt: RegisteredPrompt = {
       description,
       argsSchema: argsSchema === undefined ? undefined : z.object(argsSchema),
       callback: cb,
+      enabled: true,
+      disable: () => registeredPrompt.update({ enabled: false }),
+      enable: () => registeredPrompt.update({ enabled: true }),
+      remove: () => registeredPrompt.update({ name: null }),
+      update: (updates) => {
+        if (typeof updates.name !== "undefined" && updates.name !== name) {
+          delete this._registeredPrompts[name]
+          if (updates.name) this._registeredPrompts[updates.name] = registeredPrompt
+        }
+        if (typeof updates.description !== "undefined") registeredPrompt.description = updates.description
+        if (typeof updates.argsSchema !== "undefined") registeredPrompt.argsSchema = z.object(updates.argsSchema)
+        if (typeof updates.callback !== "undefined") registeredPrompt.callback = updates.callback
+        if (typeof updates.enabled !== "undefined") registeredPrompt.enabled = updates.enabled
+        this.sendPromptListChanged()
+      },
     };
+    this._registeredPrompts[name] = registeredPrompt;
 
     this.setPromptRequestHandlers();
+    this.sendPromptListChanged()
+
+    return registeredPrompt
+  }
+
+  /**
+   * Checks if the server is connected to a transport.
+   * @returns True if the server is connected
+   */
+  isConnected() {
```

Comment:
Nit: getters are nicer for simple reads
```suggestion
  get connected() {
```